### PR TITLE
#466: fixed workspace fallback to main

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Fixes: #466 

Implements:
* added fallback to main if no workspace was found
* added several tests for mentioned bugs

### CODE REVIEW

**Review by slskiba — APPROVED** [review]
LGTM 👍

**Review by hohwille — APPROVED** [review]
@jan-vcapgemini thank you for your PR. Now reading your diff I finally understand the bug and see what went wrong. Thanks for fixing and nice that you also improved the test to cover the gap. Great work 👍

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java [hunk 1]
```diff
@@ -178,6 +178,11 @@ public AbstractIdeContext(IdeLogLevel minLogLevel, Function<IdeLogLevel, IdeSubL
       currentDir = getParentPath(currentDir);
     }
 
+    // fallback to main
+    if (workspace.isEmpty()) {
+      workspace = WORKSPACE_MAIN;
+    }
+
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:185]: BTW: I would be a little bit more clean to solve this in line 168 like this:
```java
        if (FOLDER_WORKSPACES.equals(name1) && !name2.isEmpty()) {
```

Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java]: I just edited this online on github. Seems cleaner to not set workspace to empty string and then reset to `main` afterwards. So we can merge and include in todays release.

**Pull request merged**